scalafix-tests/input/src/main/scala/test/escapeHatch/AnchorAnnotationMixed.scala

```diff
@@ -0,0 +1,87 @@
+/*
+rules = [
+  "class:scalafix.test.EscapeHatchDummyLinter"
+  "class:scalafix.test.EscapeHatchNoNulls"
+]
+*/
+package test.escapeHatch
+
+object AnchorAnnotationMixed {
+
+  def aDummy0(x: Option[Any]): Unit = { // assert: EscapeHatchDummyLinter
+    val bDummy = 0 // assert: EscapeHatchDummyLinter
+    val cDummy = 0 // assert: EscapeHatchDummyLinter
+    val foo = null // assert: EscapeHatchNoNulls
+  }
+
+
+  // Annotation mixed with 'scalafix:ok' anchor
+  @SuppressWarnings(Array("EscapeHatchDummyLinter"))
+  def aDummy1(x: Option[Any]): Unit = {
+    val bDummy = 0
+    val cDummy = 0 /* scalafix:ok EscapeHatchDummyLinter */ // assert: UnusedScalafixSuppression.Disable
+    val foo = null // scalafix:ok EscapeHatchNoNulls
+  }
+
+
+  // Turning ON/OFF different rule using anchor within annotated method
+  @SuppressWarnings(Array("EscapeHatchDummyLinter"))
+  def aDummy2(x: Option[Any]): Unit = {
+    val bDummy = 0
+    // scalafix:off EscapeHatchNoNulls
+    val cDummy = 0
+    val foo1 = null
+    // scalafix:on EscapeHatchNoNulls
+    val foo2 = null // assert: EscapeHatchNoNulls
+  }
+
+
+  // Overlap - annotated method enclosed by anchors
+  /* scalafix:off EscapeHatchDummyLinter */ // assert: UnusedScalafixSuppression.Disable
+  @SuppressWarnings(Array("EscapeHatchDummyLinter"))
+  def aDummy3(x: Option[Any]): Unit = {
+    val bDummy = 0
+    val cDummy = 0
+  }
+  // scalafix:on EscapeHatchDummyLinter
+  val dDummy = 0 // assert: EscapeHatchDummyLinter
+
+
+
+  // Overlap - anchors enclosed by annotated method
+  @SuppressWarnings(Array("EscapeHatchDummyLinter"))
+  def aDummy4(x: Option[Any]): Unit = {
+    /* scalafix:off EscapeHatchDummyLinter */ // assert: UnusedScalafixSuppression.Disable
+    val bDummy = 0
+    // scalafix:on EscapeHatchDummyLinter
```

Comment:
this is because the annotation takes precedence when there is overlapping. See my comment https://github.com/scalacenter/scalafix/issues/442#issuecomment-366519138

Reply:
(there is a counterexample in the comment I mentioned above)

Reply:
Yup, I got that. I think we can handle the case where the anchor range position is completely inside the annotation range position.

Reply:
Also:

```scala
  // scalafix:off EscapeHatchDummyLinter
  def aDummy = {}
  // scalafix:on EscapeHatchDummyLinter

  @SuppressWarnings(Array("EscapeHatchDummyLinter"))
  def bDummy = {}

  @SuppressWarnings(Array("EscapeHatchDummyLinter"))
  def cDummy = {}
```

Could have the same effect as:

```scala
  // scalafix:off EscapeHatchDummyLinter
  def aDummy = {}
  // scalafix:on EscapeHatchDummyLinter

  // scalafix:off EscapeHatchDummyLinter
  def bDummy = {}
  // scalafix:on EscapeHatchDummyLinter

  // scalafix:off EscapeHatchDummyLinter
  def cDummy = {}
  // scalafix:on EscapeHatchDummyLinter
```

Reply:
ok, let me put some thoughts on this. @olafurpg, would you like to chip in?

Reply:
I agree with @marcelocenerine 's assessment in https://github.com/scalacenter/scalafix/issues/442#issuecomment-363778884 that it is counter-intuitive if annotations are treated as comments. Annotations should have push/pop semantics and they should override comments.

Reply:
so should I keep the current behavior then? I updated the docs last night.

Reply:
That is definitely doable. The only thing is that it would require the comment type (on+off/ok) to be preserved. Currently that information gets lost very early as `// scalafix:ok` is transformed into a pair of off/on `EscapeFilter`s. If you think this is a nice to have, then I can invest some time on it.